apps/console/src/features/users/components/user-profile.tsx

```diff
@@ -475,13 +475,18 @@ export const UserProfile: FunctionComponent<UserProfilePropsInterface> = (
     };
 
     /**
-     * This function returns the username of the current user.
+     * This function returns the username or the default email of the current user (if the username is empty).
      *
      * @param user - user that the username will be extracted from.
+     * @param setDefaultEmail - whether to return the default email address when the username is empty.
      */
-    const resolveUsername = (user: ProfileInfoInterface): string => {
+    const resolveUsernameOrDefaultEmail = (user: ProfileInfoInterface, setDefaultEmail: boolean): string => {
```

Comment:
WDYT about using a term that indicates a flag.

```js
shouldReturnDefaultEmailAsFallback
```

Reply:
Will use the suggested term. Thanks.

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/3643/commits/30ddab49ee1262ac43af3c0acf870465d9ce2766.